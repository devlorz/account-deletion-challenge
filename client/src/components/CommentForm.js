import PropTypes from 'prop-types'
import React from 'react'

const CommentForm = ({
  isFocusCommentBox,
  onChangeComment,
  comment,
  showCommentForm,
}) =>
  !showCommentForm ? null : (
    <div style={{ marginTop: '2rem' }}>
      Comments:
      <div>
        <textarea
          type="text"
          name="comment"
          style={
            isFocusCommentBox
              ? { border: '1px solid blue' }
              : { border: '1px solid black' }
          }
          onChange={onChangeComment}
          value={comment}
        />
      </div>
    </div>
  )

CommentForm.PropTypes = {
  isFocusCommentBox: PropTypes.bool,
  onChangeComment: PropTypes.func,
  comment: PropTypes.string,
  showCommentForm: PropTypes.bool,
}

export default CommentForm
