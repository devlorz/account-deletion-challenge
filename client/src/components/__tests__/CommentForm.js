import React from 'react'
import { render } from 'react-testing-library'

import CommentForm from '../CommentForm'

test('snapshot', () => {
  const { container } = render(
    <CommentForm
      isFocusCommentBox={true}
      onChangeComment={jest.fn()}
      comment={'Comment Text'}
      showCommentForm
    />
  )
  expect(container.firstChild).toMatchSnapshot()
})
