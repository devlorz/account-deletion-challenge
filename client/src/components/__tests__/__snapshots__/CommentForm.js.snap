// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  data-reactroot=""
  style="margin-top: 2rem;"
>
  <!-- react-text: 2 -->
  Comments:
  <!-- /react-text -->
  <div>
    <textarea
      name="comment"
      style="border: 1px solid blue;"
      type="text"
    >
      Comment Text
    </textarea>
  </div>
</div>
`;
